Components/Hlms/Pbs/src/Cubemaps/OgreCubemapProbe.cpp

```diff
@@ -409,27 +434,32 @@ namespace Ogre
             mCamera->setLightCullingVisibility( true, true );
         }
 
+        mTimeSlicing = ts;
+
+        mWorkspaceExecMask = executionMask;
         mWorkspaceMipmapsExecMask = mipmapsExecutionMask;
 
         if( !mCreator->getAutomaticMode() )
             mTexture->_transitionTo( GpuResidency::Resident, (uint8*)0 );
-        else
-        {
-            if( mCreator->getUseDpm2DArray() )
-                mipmapsExecutionMask = ~mipmapsExecutionMask;
-            else
-                mipmapsExecutionMask = 0xff;
-        }
 
         CompositorChannelVec channels( 1, rtt );
-        mWorkspace = compositorManager->addWorkspace( sceneManager, channels, mCamera,
-                                                      mWorkspaceDefName, false, -1,
-                                                      (UavBufferPackedVec*)0,
-                                                      (const ResourceLayoutMap*)0,
-                                                      (const ResourceAccessMap*)0,
-                                                      Vector4::ZERO,
-                                                      0x00, mipmapsExecutionMask );
-        mWorkspace->setListener( mCreator );
+        for( size_t i=0; i<6; ++i )
+        {
+            uint32 workspaceExecutionMask = ( ExecutionFlags::FIRST_SLICE_EXECUTION_FLAG << i ) |
+                                            mWorkspaceExecMask & ExecutionFlags::RESERVED_EXECUTION_FLAGS;
```

Comment:
Turn on the `ExecutionFlags::FIRST_SLICE_EXECUTION_FLAG` bit (shifted left by _i_ positions where _i_ is the cubemap face index) and turn off all the other bits internally used by OGRE

Reply:
Code has slightly changed:
```
uint32 workspaceExecutionMask = mWorkspaceExecMask & ExecutionFlags::RESERVED_EXECUTION_FLAGS;
if( i == 6 )
	workspaceExecutionMask |= ExecutionFlags::TS_IBL_EXECUTION_FLAG;
else
	workspaceExecutionMask |= ExecutionFlags::TS_FIRST_FACE_EXECUTION_FLAG << i;
```
Turn on the `ExecutionFlags::TS_FIRST_FACE_EXECUTION_FLAG` bit (shifted left by _i_ positions where _i_ is the cubemap face index) or the `ExecutionFlags::TS_IBL_EXECUTION_FLAG` bit and  combine it with `mWorkspaceExecMask `.